### PR TITLE
Acceptance test for static tags

### LINKED ISSUE #33443: Acceptance tests static tags
For https://github.com/owncloud/core/pull/33118

### PR DESCRIPTION
**Pull request by sharidas**
Acceptance test added for static tags

Signed-off-by: Sujith H <sharidasan@owncloud.com>

## Description

Add acceptance test for the static tags

## Related Issue

- Fixes https://github.com/owncloud/core/issues/33443

## Motivation and Context

Acceptance test were not added for the static tags added recently. So this patch is adding acceptance test and hence changes required for webdav. 

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

**Comment by sharidas** [comment]
Context for this change:
A new tag is added, known as static tags. The acceptance test were not added/updated when the static tags work was done. Hence this PR includes the changes to add acceptance test + changes required for the webdav.

Below is the checklist which this PR should address:
- [ ] Add acceptance test for assign for static tags.
- [ ] Add acceptance test for unassign for static tags.
- [ ] Add acceptance test for edit for static tags.
- [ ] Add acceptance test creating static tags.
- [ ] Add acceptance test for deleting static tags.
- [ ] Updates made for WebDAV code.
- [ ] phpunit test updates.
- [ ] webui acceptance test for static tags.

**Comment by sharidas** [comment]
Current status of the PR:
- Have added the code mentioned in the checklist https://github.com/owncloud/core/pull/33517#issuecomment-438653000

**Comment by PVince81** [comment]
@sharidas seems this is not only adding tests but also fixing some issues ? would be good to have this reflected in the PR subject as information, else this would get deleted from changelog

**Review by PVince81 — COMMENTED** [review]
the code changes that fixes additional issues and their unit tests look good.

would be good if @phil-davis could review the tests part

**Comment by individual-it** [comment]
acceptance tests look good to me

**Comment by sharidas** [comment]
@phil-davis @individual-it below are the changes made as per the comments

- [x] Updated the scenario with `And the administrator has created a "static" tag with name "StaticTag" and groups "group1"`
- [x] Adjusted the table columns to line up

Another round of review required for the changes updated in the feature file of acceptance tests.

File: tests/acceptance/features/apiTags/editTags.feature [hunk 1]
```diff
@@ -25,6 +25,13 @@ Feature: Editing the tags
     Then the following tags should exist for the administrator
       | JustARegularTagName | not user-assignable |
 
+  Scenario: Renaming a static tag as regular user should fail
+    Given the administrator has created a "static" tag with name "StaticTagName"
+    When the user edits the tag with name "StaticTagName" and sets its name to "AnotherTagName" using the WebDAV API
+    Then the following tags should exist for the administrator
+      | StaticTagName | static |
+
```
Comment (individual-it) [inline tests/acceptance/features/apiTags/editTags.feature]: one bonus empty line

**Comment by individual-it** [comment]
acceptance tests look very good to me :+1:

**Review by phil-davis — COMMENTED** [review]
I agree, acceptance tests are good now.
@PVince81 or @DeepDiver1975 please approve and merge when you are happy with the rest of the code.

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
@sharidas please backport

**Pull request merged**